pimd/pim_cmd.c

```diff
@@ -2697,6 +2697,72 @@ static void pim_show_join_desired(struct pim_instance *pim, struct vty *vty,
 		vty_json(vty, json);
 }
 
+static void pim_groups_show_information(struct pim_instance *pim,
+					struct vty *vty, bool uj)
+{
+	struct interface *ifp;
+	json_object *json = NULL;
+	json_object *json_group = NULL;
+	json_object *json_groups = NULL;
+
+	if (uj) {
+		json = json_object_new_object();
+		json_groups = json_object_new_array();
+		json_object_object_add(json, "groups", json_groups);
+	} else {
+		vty_out(vty, "Group           RP              Origin  Mode\n");
+	}
+
+	FOR_ALL_INTERFACES (pim->vrf, ifp) {
+		struct pim_interface *pim_ifp = ifp->info;
+		struct listnode *grpnode;
+		struct gm_group *grp;
+
+		if (!pim_ifp)
+			continue;
+
+		for (ALL_LIST_ELEMENTS_RO(pim_ifp->gm_group_list, grpnode,
+					  grp)) {
+			struct rp_info *rp_info = NULL;
+			struct prefix group_prefix;
+			const char *origin = NULL;
+			const char *mode = NULL;
+
+			pim_addr_to_prefix(&group_prefix, grp->group_addr);
+			rp_info = pim_rp_find_match_group(pim, &group_prefix);
+
+			origin = pim_rp_origin_str(rp_info->rp_src);
+			mode = pim_is_grp_ssm(pim, rp_info->group.u.prefix4)
+				       ? "SSM"
+				       : "ASM";
+
+			if (uj) {
```

Comment:
it's completely possible to have the same group incoming on multiple interfaces.  Given the output, I don't see much need to print the same one multiple times.

Reply:
Do you mean including "interface" is enough?
```
"groups":[
    {
      "group":"238.0.0.5",
      "rp":"192.168.10.17",
      "origin":"Static",
      "mode":"ASM",
      "interface": "eth1"
    },
    {
      "group":"238.0.0.5",
      "rp":"192.168.11.17",
      "origin":"Static",
      "mode":"ASM",
      "interface": "eth2"
    },
  ]
```

Reply:
What's the point of the command here?  Why would the end operator want to see the same group across all the interfaces?

Reply:
I'm just trying to understand which format do you have in mind.